### DISCUSSION TITLE
Access the color palette?

### DISCUSSION THREAD

**Discussion by guerda** [vasturiano/force-graph#349] [Q&A] [answered]
I have an attribute defined to auto-color the nodes based on a grouping attribute.
I would like to add a simple legend with colors and the grouping attribute.

In d3.js I did the following thing, but I seem to miss how I can access force-graph's color palette. Is there any way to do this?

    var color = d3.scaleOrdinal(d3.schemeCategory10);
    for(int i = 0; i < ...) {
       color(i);
    }

**Comment by vasturiano** [comment, marked as answer]
@guerda the auto-color scale is just a quick convenience method for assigning different colors based on a simple logic. If you need anything more sophisticated than that, it's better you use the more appropriate `nodeColor` method. In fact, if you're already declaring a color scale externally, like you shown above, all you need to do is assign that color scale to the nodes. Something like:
```js
.nodeColor(color)
```
With `color` being the color scale you've defined in your code.

**Reply by guerda** [reply]
It worked, thank you. That's how I did it now:

    .nodeColor(node` => getColorForPhase([node.rdphase]))
